components/tools/OmeroPy/test/integration/gatewaytest/test_permissions.py

```diff
@@ -43,3 +43,25 @@ def test_update_admin_privileges(self):
 
         expected = ['Sudo', 'ModifyGroup', 'ModifyUser']
         assert set(expected) == set(conn.getAdminPrivileges(exp.id.val))
+
+    def test_full_admin_privileges(self):
+        """Test full admin privileges check."""
+        conn = BlitzGateway(client_obj=self.root, try_super=True)
+        allPrivs = []
+        for p in conn.getEnumerationEntries('AdminPrivilege'):
+            allPrivs.append(p.getValue())
+        conn.updateAdminPrivileges(self.user.id.val, add=allPrivs)
+        test = conn.getCurrentAdminPrivileges()
+        assert set(allPrivs) == set(test)
+        assert conn.isFullAdmin()
+
+        conn.updateAdminPrivileges(self.user.id.val, remove=['Chown'])
+
+        privs = set(conn.getCurrentAdminPrivileges())
```

Comment:
User still has all privileges, including 'Chown'. Does anyone know why?

Reply:
In the same session you have to wait a minute for the cache to expire. If they have a new session you should see the change immediately.

Reply:
Perhaps because https://github.com/openmicroscopy/openmicroscopy/blob/develop/components/server/src/ome/logic/AdminImpl.java#L1622 doesn't refresh the event context? If so, then after https://github.com/openmicroscopy/openmicroscopy/blob/develop/components/server/src/ome/logic/AdminImpl.java#L1359 we should do so.

You can test the hypothesis by adding a call to [IAdmin.synchronizeLoginCache](https://github.com/openmicroscopy/openmicroscopy/blob/develop/components/server/src/ome/logic/AdminImpl.java#L415) between your `update` and your `get`

Reply:
:-1: for introducing a `sleep()`

Reply:
Should also work fine to just relogin in, whatever the Python test equivalent is of `loginUser(eventContext)` is.

Reply:
Agreed, but it'd be nice to solve this server side so that it's not necessary for clients to jump through hoops.

Reply:
I tried both, wait and call `synchronizeLoginCache` neither did work. I'll check if I can somehow re-login in the test.

Reply:
@dominikl : and an explicit call to `getAdminService().getEventContext()`?

Reply:
Ok, I guess one problem is that I create the `conn` with `client_obj=self.root`...  I just remove the test, maybe someone else can add one. Can't get my head around this `self.*` stuff, `self.root`, `self.client`, `self.user`, ... Makes me really appreciate real OO typed languages.